### PR TITLE
#86 Update frameworks and packages

### LINKED ISSUE #86: Update frameworks and packages
- [x] Update the NestJS version and the libraries from the transcendence-app package.json
- [x] Update Dockerfiles and docker-compose.yml files to use the latest LTS node version and docker images
- [x] Update GitHub actions node version

### PR DESCRIPTION
**Pull request by gbudau**
Please take these steps, to use the same node, npm, and nestjs CLI versions:

Install and use node v16.16.0
```
nvm install v16.16.0
nvm use v16.16.0
```

Update the @nestjs/cli package
```
npm update -g @nestjs/cli
```

Redeploy Docker services and update images
```
docker compose up -d --build
```

Install the node packages
```
cd transcendence-app
npm ci
cd ../webapp
npm ci
cd ..
```

closes #86 

---

- [x] Update Dockerfiles
- [x] Update docker-compose files
- [x] Update Github actions
- [x] Update transcendence-app packages
- [x] Update webapp packages

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ services:
       - "memcached"
 
   db:
-    image: postgres:14.2-alpine3.15
+    image: postgres:14.4-alpine3.16
```
Comment (Damrod) [inline docker-compose.yml:23]: Is this completely stable?
In my experience,  `[3.16.0, 3.16, 3, latest](https://github.com/alpinelinux/docker-alpine/blob/c9515859c035acbb899c6738506954e36f57e7dc/x86_64/Dockerfile)[3.16.0, 3.16, 3, latest](https://github.com/alpinelinux/docker-alpine/blob/c9515859c035acbb899c6738506954e36f57e7dc/x86_64/Dockerfile)` (at this date, 3.16 seems to be latest) gives problems.

**Review by Damrod — APPROVED** [review]
Read the comment, but anyway if it's good enough in your opinion, here you have my approval.

File: docker-compose.yml (hunk 1 above)
Comment (gbudau) [reply docker-compose.yml:23]: It was released two months ago, and from what I've seen on DockerHub, all the popular images are starting to use this version for their alpine image version. But if we find any issues, we can downgrade later.

**Comment by gbudau** [comment]
Thank you! :rocket:

**Pull request merged**